rust/lance-namespace/src/dir.rs

```diff
@@ -0,0 +1,674 @@
+                storage_options.insert(storage_key.to_string(), value.clone());
+            }
+        }
+        storage_options
+    }
+    
+    pub fn root(&self) -> Option<String> {
+        self.root.clone()
+    }
+    
+    pub fn storage_options(&self) -> &HashMap<String, String> {
+        &self.storage_options
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::TempDir;
+
+    #[tokio::test]
+    async fn test_dir_namespace_creation() {
+        let temp_dir = TempDir::new().unwrap();
+        let root_path = temp_dir.path().to_string_lossy().to_string();
+        
+        let mut properties = HashMap::new();
+        properties.insert("root".to_string(), root_path);
+        
+        let namespace = DirNamespace::new(properties).unwrap();
+        
+        // Test list_tables on empty directory
+        let request = ListTablesRequest {
+            id: None,
+            page_token: None,
+            limit: None,
+        };
+        
+        let response = namespace.list_tables(request).await.unwrap();
+        assert!(response.tables.is_empty());
+    }
+
+    #[tokio::test]
+    async fn test_namespace_operations_not_supported() {
+        let temp_dir = TempDir::new().unwrap();
+        let root_path = temp_dir.path().to_string_lossy().to_string();
+        
+        let mut properties = HashMap::new();
+        properties.insert("root".to_string(), root_path);
+        
+        let namespace = DirNamespace::new(properties).unwrap();
+        
+        // Test that namespace operations are not supported
+        let create_namespace_request = CreateNamespaceRequest {
+            id: Some(vec!["test".to_string()]),
+            properties: None,
+            mode: None,
+        };
+        
+        let result = namespace.create_namespace(create_namespace_request).await;
+        assert!(result.is_err());
+        assert!(result.unwrap_err().to_string().contains("does not support"));
+        
+        // Test list_namespaces
+        let list_request = ListNamespacesRequest {
+            id: None,
+            page_token: None,
+            limit: None,
+        };
+        let result = namespace.list_namespaces(list_request).await;
+        assert!(result.is_err());
+        
+        // Test describe_namespace
+        let describe_request = DescribeNamespaceRequest {
+            id: Some(vec!["test".to_string()]),
+        };
+        let result = namespace.describe_namespace(describe_request).await;
+        assert!(result.is_err());
+        
+        // Test drop_namespace
+        let drop_request = DropNamespaceRequest {
+            id: Some(vec!["test".to_string()]),
+            mode: None,
+            behavior: None,
+        };
+        let result = namespace.drop_namespace(drop_request).await;
+        assert!(result.is_err());
+    }
+
+    #[tokio::test]
+    async fn test_table_operations() {
```

Comment:
do one test per operation for table and namespace, instead of squeezing everything together.